### DISCUSSION TITLE
How to make changes to robot.py module active?

### DISCUSSION THREAD

**Discussion by geoc1234** [NVIDIA-AI-IOT/jetbot#372] [Q&A] [answered]
I have a Nano 4gb b0.  My robot base uses an arduino, which listens for cmd_vel_msgs over ROS. i need to modify the  Robot class in the jetbot container so that it publishes ROS messages, but i don't know how to make any of the edits i make to robot.py take effect. 

am i supposed to execute /docker/build.sh after making changes?

**Comment by jaybdub** [comment, marked as answer]
Hey @geoc1234,

Thanks for reaching out!

I think the following commands should work

```bash
cd docker
./disable.sh
./configure.sh
./build.sh
./enable.sh
```

I'd recommend modifying ``JETBOT_VERSION`` in ``docker/configure.sh`` to something else like ``JETBOT_VERSION=ros_motors`` so you can keep / use the old container if needed.

Please let me know if this helps or you have any questions.

Best,
John

**Reply by geoc1234** [reply]
@jaybdub (John), thank you!

Your suggestion worked.  Although, because I modified the docker file for the base to include instructions for building ROS noetic, the build.sh script takes a long time to run because it rebuilds ROS.  Maybe I have something configured incorrectly or am not using docker correctly?  I'm not a programmer, but isn't python an interpretive language.  If so, why wouldn't changes to the source *.py module become active without a build? 

Again, thanks so much for your help.   How do I send you a coffee?
